client/authority-discovery/src/worker/tests.rs

```diff
@@ -462,33 +432,29 @@ fn dont_stop_polling_when_error_is_returned() {
 	let mut pool = LocalPool::new();
 
 	let (_to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
+	let worker = Worker::new(
 		from_service,
 		test_api,
 		network.clone(),
 		vec![],
 		dht_event_rx.boxed(),
-		Role::Authority(key_store),
+		Role::Authority(Arc::new(key_store)),
 		None,
 	);
 
 	// Spawn the authority discovery to make sure it is polled independently.
 	//
 	// As this is a local pool, only one future at a time will have the CPU and
 	// can make progress until the future returns `Pending`.
-	pool.spawner().spawn_local_obj(
-		futures::future::poll_fn(move |ctx| {
-			match std::pin::Pin::new(&mut worker).poll(ctx) {
-				Poll::Ready(()) => {},
-				Poll::Pending => {
-					discovery_update_tx.send(Event::Processed).now_or_never();
-					return Poll::Pending;
-				},
-			}
-			let _ = discovery_update_tx.send(Event::End).now_or_never().unwrap();
-			Poll::Ready(())
-		}).boxed_local().into(),
-	).expect("Spawns authority discovery");
+	let _ = pool.spawner().spawn_local_obj(async move {
+		worker.run().await
+	}.boxed_local().into());
+	let _ = pool.spawner().spawn_local_obj(async move {
+		Delay::new(Duration::from_secs(1)).await;
+		discovery_update_tx.send(Event::Processed).now_or_never();
```

Comment:
```suggestion
		discovery_update_tx.send(Event::Processed).now_or_never().unwrap();
```

To make sure we don't ignore the case where sending fails.